reqs/tests/link_agencies_tests.py

```diff
@@ -0,0 +1,123 @@
+from reqs.models import Agency, AgencyGroup, Requirement
+
+
+@pytest.fixture
+def populate_agencies():
+    synch = SynchCommand()
+    synch.create_system_groups()
+    synch.sync_row({
+        'agencyAbbreviation': None,
+        'agencyCode': '123',
+        'agencyName': 'Aquarius',
+        'agencyType': '5-Other Branches',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "DOD",
+        'agencyCode': '111',
+        'agencyName': 'Department of Defense',
+        'agencyType': '',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "DHS",
+        'agencyCode': '100',
+        'agencyName': 'Department of Homeland Security',
+        'agencyType': '',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "GSA",
+        'agencyCode': '101',
+        'agencyName': 'General Services Administration',
+        'agencyType': '',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "Justice",
+        'agencyCode': '102',
+        'agencyName': 'Department of Justice',
+        'agencyType': '',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "FOO",
+        'agencyCode': '112',
+        'agencyName': 'Department of Foo',
+        'agencyType': '',
+        'CFO_Act': None,
+        'CIO_Council': None,
+    })
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("impacted_entity,agencies,agency_groups", [
+    (("",), (set(),), (set(),)),
+    (("All agencies, Defense",), ({"DOD"},), ({"all-agencies"},)),
+    (("dhs, Defense",), ({"DHS", "DOD"},), (set(),)),
+    (("CFO, Defense",), ({"DOD"},), ({"cfo-act"},)),
+    (("CFO; Defense; all agencies",), ({"DOD"},),
+     ({"all-agencies", "cfo-act"},)),
+    (("All agencies", "DOD"), (set(), {"DOD"}), ({"all-agencies"}, set())),
+    (("CFO\nDHS\nGSA\nDOJ\nDOD",), ({"DHS", "DOD", "Justice", "GSA"},),
+     ({"cfo-act"},)),
+    (("All agencies\nDHS\nGSA\nDOJ\nDOD",),
+     ({"DHS", "DOD", "Justice", "GSA"},), ({"all-agencies"},)),
+    (("DHS\nGSA\nDOJ\nDOD",), ({"DHS", "DOD", "Justice", "GSA"},),
+     (set(),)),
+    (("Federal CFO Council, DHS\nGSA\nDOJ\nDOD",),
+     ({"DHS", "DOD", "Justice", "GSA"},), (set(),)),
+    (("All agencies except micro agencies\nDHS\nGSA\nDOJ\nDOD",),
+     ({"DHS", "DOD", "Justice", "GSA"},), (set(),)),
+    (("All agencies except Paperwork\nDHS\nGSA\nDOJ\nDOD",),
+     ({"DHS", "DOD", "Justice", "GSA"},), (set(),)),
+    (("department of homeland\ngeneral services\nJustice\nDefense",),
+     ({"DHS", "DOD", "Justice", "GSA"},), (set(),)),
+], ids=repr)
+def test_linking(populate_agencies, impacted_entity, agencies,
+                 agency_groups):
+    reqs = [mommy.make(Requirement, impacted_entity=_) for _ in
+            impacted_entity]
```

Comment:
I don't mind changing it, but there's a philosophical disagreement here in that I think it means “won't be using the variable outside the current context”…